src/pages/DataAccessRequestApplication_new.js

```diff
@@ -290,139 +317,245 @@ const DataAccessRequestApplicationNew = (props) => {
+    const userId = Storage.getCurrentUser().userId;
+    let formattedFormData = cloneDeep(formData);
+
+    for (var key in formattedFormData) {
+      if (formattedFormData[key] === '') {
+        formattedFormData[key] = undefined;
+      }
+    }
+    formattedFormData.userId = userId;
+
+    try {
+      let referenceId = formattedFormData.referenceId;
+      let darPartialResponse = await updateDraftResponse(formattedFormData, referenceId);
+      referenceId = darPartialResponse.referenceId;
+
+      if(!isNil(uploadedIrbDocument) || !isNil(uploadedCollaborationLetter)) {
+        darPartialResponse = await saveDARDocuments(uploadedIrbDocument, uploadedCollaborationLetter, referenceId);
+      }
+      let updatedFormData = assign(formattedFormData, darPartialResponse);
+      await DAR.postDar(updatedFormData);
+      setShowDialogSubmit({
+        showDialogSubmit: false
+      }, Navigation.console(Storage.getCurrentUser(), props.history).response);
+    } catch (error) {
+      setShowDialogSubmit(false);
+      NotyUtil.showError({
+        text: 'Error: Data Access Request submission failed'
+      });
+    }
+  };
+
+  const onSaveConfirmation = (selectedOk) => () => {
+    setDisableOkButton(true);
+    if (selectedOk === true) {
+      saveDarDraft();
+      setDisableOkButton(false);
+    } else {
+      setShowDialogSave(false);
+      setDisableOkButton(false);
+    }
+  };
+
+  const onSubmitConfirmation = (selectedOk) => () => {
+    setDisableOkButton(true);
+    if (selectedOk === true) {
+      submitDARFormData();
+      setDisableOkButton(false);
+    } else {
+      setShowDialogSubmit(false);
+      setDisableOkButton(false);
+    }
+  };
+
+
+  const saveDarDraft = async () => {
+    let formattedFormData = cloneDeep(formData);
+    // DAR datasetIds needs to be a list of ids
+
+    // Make sure we navigate back to the current DAR after saving.
+    const { dataRequestId } = props.match.params;
+    try {
+      let referenceId = formattedFormData.referenceId;
+
+      let darPartialResponse = await updateDraftResponse(formattedFormData, referenceId);
+      referenceId = darPartialResponse.referenceId;
+      if(isNil(dataRequestId)) {
+        props.history.replace('/dar_application/' + referenceId);
+      }
+      //execute saveDARDocuments method only if documents are required for the DAR
+      //value can be determined from activeDULQuestions, which is populated on Step 2 where document upload occurs
+      if(!isNil(uploadedIrbDocument) || !isNil(uploadedCollaborationLetter)) {
+        darPartialResponse = await saveDARDocuments(uploadedIrbDocument, uploadedCollaborationLetter, referenceId);
+      }
+      batchFormFieldChange(darPartialResponse);
+      setShowDialogSave(false);
+      setDisableOkButton(false);
+    } catch(error) {
+      setShowDialogSave(false);
+      setDisableOkButton(false);
+      NotyUtil.showError('Error saving partial Data Access Request update');
```

Comment:
You might want to include what the user should do at this point?  Try saving the request again?